### ISSUE TITLE
Error running neuroscout v0.6 singularity

### ISSUE THREAD

**Issue by jsmentch** [neuroscout/neuroscout#1035]
I was trying to run an analysis but running into an error.

I'm using a singularity image of v0.6 of neuroscout-cli (from dockerhub, I tried out singularity pull oras://ghcr.io/neuroscout/neuroscout-cli:master but that wasn't working)

my command:

```
singularity run --cleanenv -B /nobackup \
-B /nobackup/scratch/Fri/jsmentch/feature_viz_paper/working:/workdir \
-B /nobackup/scratch/Fri/jsmentch/feature_viz_paper/home:/home \
-B /nobackup/scratch/Fri/jsmentch/feature_viz_paper/tmp:/tmp \
/nobackup/scratch/Fri/jsmentch/feature_viz_paper/neuroscout.simg \
run \
--n-cpus=8 \
--download-dir=/nobackup/scratch/Fri/jsmentch/feature_viz_paper/cache \
f9uii \
/nobackup/scratch/Fri/jsmentch/feature_viz_paper/outputs
```

Anyone know what might be going wrong or should I maybe use a different version? Error below:

```
Traceback (most recent call last):
  File "/opt/miniconda-latest/envs/neuro/lib/python3.9/site-packages/nipype/pipeline/plugins/multiproc.py", line 67, in run_node
    result["result"] = node.run(updatehash=updatehash)
  File "/opt/miniconda-latest/envs/neuro/lib/python3.9/site-packages/nipype/pipeline/engine/nodes.py", line 521, in run
    result = self._run_interface(execute=True)
  File "/opt/miniconda-latest/envs/neuro/lib/python3.9/site-packages/nipype/pipeline/engine/nodes.py", line 639, in _run_interface
    return self._run_command(execute)
  File "/opt/miniconda-latest/envs/neuro/lib/python3.9/site-packages/nipype/pipeline/engine/nodes.py", line 750, in _run_command
    raise NodeExecutionError(
nipype.pipeline.engine.nodes.NodeExecutionError: Exception raised while executing Node loader.

Traceback (most recent call last):
  File "/opt/miniconda-latest/envs/neuro/lib/python3.9/site-packages/nipype/interfaces/base/core.py", line 398, in run
    runtime = self._run_interface(runtime)
  File "/opt/miniconda-latest/envs/neuro/lib/python3.9/site-packages/fitlins/interfaces/bids.py", line 250, in _run_interface
    self._results['all_specs'] = self._load_graph(runtime, graph)
  File "/opt/miniconda-latest/envs/neuro/lib/python3.9/site-packages/fitlins/interfaces/bids.py", line 258, in _load_graph
    specs = node.run(inputs, group_by=node.group_by, **filters)
  File "/opt/miniconda-latest/envs/neuro/lib/python3.9/site-packages/bids/modeling/statsmodels.py", line 445, in run
    groups = self._build_groups(all_objects, group_by)
  File "/opt/miniconda-latest/envs/neuro/lib/python3.9/site-packages/bids/modeling/statsmodels.py", line 348, in _build_groups
    raise ValueError("group_by contains variable(s) {} that could not "
ValueError: group_by contains variable(s) ['run'] that could not be found in the entity index.
```

**Comment by adelavega** [comment]
The likely issue is this: we upgraded from the previoys BIDS StatsModel version to v1, but neuroscout frontend still writes out the old version. I wrote a script to convert them at run time, but it's possible that some variations fail this conversion.

You should be able to look at the output directory and see the original model, and one ending in `_v1.json` that is the converted version. Paste them here if you find it.

**Comment by jsmentch** [comment]
the id is f9uii and made via the web app:

model.json

`{"Name": "featviz_merlin_any_faces_edited", "Input": {"Task": "MerlinMovie", "Subject": ["25", "20", "21", "22", "23", "24", "26", "27", "28", "29", "30", "31", "32", "33", "34", "35", "36", "19"]}, "Steps": [{"Level": "Run", "Model": {"X": ["any_faces_edited", "rot_x", "rot_y", "rot_z", "trans_x", "trans_y", "trans_z", "speech_edited", "framewise_displacement", "csf", "a_comp_cor_00", "a_comp_cor_01", "a_comp_cor_02", "a_comp_cor_03", "a_comp_cor_04", "a_comp_cor_05", "cosine00", "cosine01", "cosine02", "cosine03", "cosine04", "cosine05", "cosine06", "cosine07", "cosine08", "cosine09", "cosine10", "cosine11", "cosine12", "cosine13", "cosine14", "cosine15", "cosine16", "cosine17", "cosine18", "cosine19", "cosine20", "cosine21", "cosine22", "motion_outlier19", "motion_outlier20", "motion_outlier21", "motion_outlier22", "motion_outlier23", "motion_outlier24", "motion_outlier25", "motion_outlier26", "motion_outlier27", "motion_outlier28", "motion_outlier29", "motion_outlier30", "motion_outlier31", "motion_outlier32", "motion_outlier33", "motion_outlier34", "motion_outlier35", "motion_outlier36", "motion_outlier37", "motion_outlier38", "motion_outlier39", "motion_outlier40", "motion_outlier41", "motion_outlier42", "motion_outlier43", "motion_outlier44", "motion_outlier45", "motion_outlier46", "motion_outlier47", "motion_outlier48", "motion_outlier49", "motion_outlier50", "motion_outlier51", "motion_outlier52", "motion_outlier53", "motion_outlier54", "motion_outlier55", "motion_outlier56", "motion_outlier57", "motion_outlier58", "motion_outlier59", "motion_outlier60", "motion_outlier61", "motion_outlier62", "motion_outlier63", "motion_outlier64", "motion_outlier65", "motion_outlier66", "motion_outlier67", "motion_outlier68", "motion_outlier69", "motion_outlier70", "motion_outlier71", "motion_outlier72", "motion_outlier73", "motion_outlier74", "motion_outlier75", "motion_outlier76", "motion_outlier77", "motion_outlier78", "motion_outlier79", "motion_outlier80", "motion_outlier81", "motion_outlier82", "motion_outlier83", "motion_outlier84", "motion_outlier85", "motion_outlier86", "motion_outlier87", "motion_outlier88", "motion_outlier89", "motion_outlier90", "motion_outlier91", "motion_outlier92", "motion_outlier93", "motion_outlier94", "motion_outlier95", "motion_outlier96", "motion_outlier97", "motion_outlier98", "motion_outlier99", "motion_outlier00", "motion_outlier01", "motion_outlier02", "motion_outlier03", "motion_outlier04", "motion_outlier05", "motion_outlier06", "motion_outlier07", "motion_outlier08", "motion_outlier09", "motion_outlier10", "motion_outlier100", "motion_outlier101", "motion_outlier102", "motion_outlier103", "motion_outlier104", "motion_outlier105", "motion_outlier106", "motion_outlier107", "motion_outlier108", "motion_outlier109", "motion_outlier11", "motion_outlier110", "motion_outlier111", "motion_outlier112", "motion_outlier113", "motion_outlier114", "motion_outlier115", "motion_outlier116", "motion_outlier117", "motion_outlier118", "motion_outlier119", "motion_outlier12", "motion_outlier120", "motion_outlier121", "motion_outlier122", "motion_outlier123", "motion_outlier124", "motion_outlier125", "motion_outlier126", "motion_outlier127", "motion_outlier128", "motion_outlier129", "motion_outlier13", "motion_outlier130", "motion_outlier131", "motion_outlier132", "motion_outlier133", "motion_outlier134", "motion_outlier135", "motion_outlier136", "motion_outlier137", "motion_outlier138", "motion_outlier139", "motion_outlier14", "motion_outlier140", "motion_outlier141", "motion_outlier142", "motion_outlier143", "motion_outlier144", "motion_outlier145", "motion_outlier146", "motion_outlier147", "motion_outlier148", "motion_outlier149", "motion_outlier15", "motion_outlier150", "motion_outlier151", "motion_outlier152", "motion_outlier153", "motion_outlier154", "motion_outlier155", "motion_outlier156", "motion_outlier157", "motion_outlier158", "motion_outlier159", "motion_outlier16", "motion_outlier160", "motion_outlier161", "motion_outlier162", "motion_outlier163", "motion_outlier164", "motion_outlier165", "motion_outlier166", "motion_outlier167", "motion_outlier168", "motion_outlier169", "motion_outlier17", "motion_outlier170", "motion_outlier171", "motion_outlier172", "motion_outlier173", "motion_outlier174", "motion_outlier175", "motion_outlier176", "motion_outlier177", "motion_outlier178", "motion_outlier179", "motion_outlier18", "motion_outlier180"]}, "Contrasts": [{"Name": "speech_edited", "Type": "t", "Weights": [1], "ConditionList": ["speech_edited"]}, {"Name": "any_faces_edited", "Type": "t", "Weights": [1], "ConditionList": ["any_faces_edited"]}], "Transformations": [{"Name": "Convolve", "Input": ["any_faces_edited", "speech_edited"]}]}, {"Level": "Subject", "DummyContrasts": {"Type": "FEMA"}}, {"Level": "Dataset", "DummyContrasts": {"Type": "t"}}], "Description": "dummy contrasts of a cleaned any_faces feature and speech feature from merlin"}`


model_v1.json

`{"Name": "featviz_merlin_any_faces_edited", "Input": {"Task": "MerlinMovie", "Subject": ["25", "20", "21", "22", "23", "24", "26", "27", "28", "29", "30", "31", "32", "33", "34", "35", "36", "19"]}, "Description": "dummy contrasts of a cleaned any_faces feature and speech feature from merlin", "BIDSModelVersion": 1.0, "Nodes": [{"Level": "run", "Model": {"X": ["any_faces_edited", "rot_x", "rot_y", "rot_z", "trans_x", "trans_y", "trans_z", "speech_edited", "framewise_displacement", "csf", "a_comp_cor_00", "a_comp_cor_01", "a_comp_cor_02", "a_comp_cor_03", "a_comp_cor_04", "a_comp_cor_05", "cosine00", "cosine01", "cosine02", "cosine03", "cosine04", "cosine05", "cosine06", "cosine07", "cosine08", "cosine09", "cosine10", "cosine11", "cosine12", "cosine13", "cosine14", "cosine15", "cosine16", "cosine17", "cosine18", "cosine19", "cosine20", "cosine21", "cosine22", "motion_outlier19", "motion_outlier20", "motion_outlier21", "motion_outlier22", "motion_outlier23", "motion_outlier24", "motion_outlier25", "motion_outlier26", "motion_outlier27", "motion_outlier28", "motion_outlier29", "motion_outlier30", "motion_outlier31", "motion_outlier32", "motion_outlier33", "motion_outlier34", "motion_outlier35", "motion_outlier36", "motion_outlier37", "motion_outlier38", "motion_outlier39", "motion_outlier40", "motion_outlier41", "motion_outlier42", "motion_outlier43", "motion_outlier44", "motion_outlier45", "motion_outlier46", "motion_outlier47", "motion_outlier48", "motion_outlier49", "motion_outlier50", "motion_outlier51", "motion_outlier52", "motion_outlier53", "motion_outlier54", "motion_outlier55", "motion_outlier56", "motion_outlier57", "motion_outlier58", "motion_outlier59", "motion_outlier60", "motion_outlier61", "motion_outlier62", "motion_outlier63", "motion_outlier64", "motion_outlier65", "motion_outlier66", "motion_outlier67", "motion_outlier68", "motion_outlier69", "motion_outlier70", "motion_outlier71", "motion_outlier72", "motion_outlier73", "motion_outlier74", "motion_outlier75", "motion_outlier76", "motion_outlier77", "motion_outlier78", "motion_outlier79", "motion_outlier80", "motion_outlier81", "motion_outlier82", "motion_outlier83", "motion_outlier84", "motion_outlier85", "motion_outlier86", "motion_outlier87", "motion_outlier88", "motion_outlier89", "motion_outlier90", "motion_outlier91", "motion_outlier92", "motion_outlier93", "motion_outlier94", "motion_outlier95", "motion_outlier96", "motion_outlier97", "motion_outlier98", "motion_outlier99", "motion_outlier00", "motion_outlier01", "motion_outlier02", "motion_outlier03", "motion_outlier04", "motion_outlier05", "motion_outlier06", "motion_outlier07", "motion_outlier08", "motion_outlier09", "motion_outlier10", "motion_outlier100", "motion_outlier101", "motion_outlier102", "motion_outlier103", "motion_outlier104", "motion_outlier105", "motion_outlier106", "motion_outlier107", "motion_outlier108", "motion_outlier109", "motion_outlier11", "motion_outlier110", "motion_outlier111", "motion_outlier112", "motion_outlier113", "motion_outlier114", "motion_outlier115", "motion_outlier116", "motion_outlier117", "motion_outlier118", "motion_outlier119", "motion_outlier12", "motion_outlier120", "motion_outlier121", "motion_outlier122", "motion_outlier123", "motion_outlier124", "motion_outlier125", "motion_outlier126", "motion_outlier127", "motion_outlier128", "motion_outlier129", "motion_outlier13", "motion_outlier130", "motion_outlier131", "motion_outlier132", "motion_outlier133", "motion_outlier134", "motion_outlier135", "motion_outlier136", "motion_outlier137", "motion_outlier138", "motion_outlier139", "motion_outlier14", "motion_outlier140", "motion_outlier141", "motion_outlier142", "motion_outlier143", "motion_outlier144", "motion_outlier145", "motion_outlier146", "motion_outlier147", "motion_outlier148", "motion_outlier149", "motion_outlier15", "motion_outlier150", "motion_outlier151", "motion_outlier152", "motion_outlier153", "motion_outlier154", "motion_outlier155", "motion_outlier156", "motion_outlier157", "motion_outlier158", "motion_outlier159", "motion_outlier16", "motion_outlier160", "motion_outlier161", "motion_outlier162", "motion_outlier163", "motion_outlier164", "motion_outlier165", "motion_outlier166", "motion_outlier167", "motion_outlier168", "motion_outlier169", "motion_outlier17", "motion_outlier170", "motion_outlier171", "motion_outlier172", "motion_outlier173", "motion_outlier174", "motion_outlier175", "motion_outlier176", "motion_outlier177", "motion_outlier178", "motion_outlier179", "motion_outlier18", "motion_outlier180", 1]}, "Contrasts": [{"Name": "speech_edited", "Type": "t", "Weights": [1], "ConditionList": ["speech_edited"], "Test": "t"}, {"Name": "any_faces_edited", "Type": "t", "Weights": [1], "ConditionList": ["any_faces_edited"], "Test": "t"}], "Name": "run", "GroupBy": ["run", "subject"], "Transformations": {"Transformer": "pybids-transforms-v1", "Instructions": [{"Name": "Convolve", "Input": ["any_faces_edited", "speech_edited"]}]}}, {"Level": "subject", "DummyContrasts": {"Test": "t"}, "Name": "subject", "GroupBy": ["subject", "contrast"], "Model": {"X": [1], "Type": "Meta"}}, {"Level": "dataset", "DummyContrasts": {"Test": "t"}, "Name": "dataset", "GroupBy": ["contrast", "group"], "Model": {"X": [1]}}]}`

**Comment by adelavega** [comment]
Oh I know what's going on. This dataset has no runs, because its one run per subject. However, it should implicitly have that, since we're still modeling "run" level estimates and then combining them. In the case where there is only 1 run per subject, the combination step doesn't do anything but copy the run level estimates over, but its still nice because it keeps the models the same. 

If you have time to play w/ this you can try editing the model. It would require either removing a level (i.e. only run level model then group level), or we need to find a way for the `groupby: ['run']` to not fail when theres is only 1 run.

**Comment by adelavega** [comment]
https://github.com/poldracklab/fitlins/issues/346

**Comment by effigies** [comment]
https://github.com/bids-standard/pybids/pull/822

**Comment by adelavega** [comment]
@jsmentch This should be fixed in FitLins/pybids, just waiting for a release so I can update neuroscout-cli

**Comment by adelavega** [comment]
@jsmentch I made a new release that fixes this: v0.6.1

Let me know if this issue crops up again
